### PR TITLE
Allow spacing in AssignAll comments

### PR DESCRIPTION
**Pull request by anserg256**
It seems that upgrading to 1.6.0 and above can break the detection of AssignAll regions. Earlier versions allow for arbitrary spacing between `//` and `AssignAll` (see 8866208). So if one has comments like this: `//AssignAll enable` and upgrades to the higher versions, the analyzer will stop working in a rather unnoticeable fashion.

I prepared a small fix for the no space case. Other possible cases with multiple spaces or tabs are not covered, but I guess they are less probable.

### CODE REVIEW

File: AssignAll/AssignAll/AssignAllMembers/RegionsToAnalyzeProvider.cs [hunk 1]
```diff
@@ -10,8 +10,8 @@ namespace AssignAll.AssignAllMembers
 {
     internal static class RegionsToAnalyzeProvider
     {
-        private static readonly Regex AssignAllDisableRegex = new Regex(@"^// AssignAll disable$", RegexOptions.Compiled | RegexOptions.IgnoreCase);
-        private static readonly Regex AssignAllEnableRegex = new Regex(@"^// AssignAll enable$", RegexOptions.Compiled | RegexOptions.IgnoreCase);
+        private static readonly Regex AssignAllDisableRegex = new Regex(@"^// ?AssignAll disable$", RegexOptions.Compiled | RegexOptions.IgnoreCase);
+        private static readonly Regex AssignAllEnableRegex = new Regex(@"^// ?AssignAll enable$", RegexOptions.Compiled | RegexOptions.IgnoreCase);
```
Comment (angularsen) [inline AssignAll/AssignAll/AssignAllMembers/RegionsToAnalyzeProvider.cs]: I would maybe just expand to `\s*` to allow any whitespace or no whitespace, both leading and trailing. In particular trailing whitespace would be easy to miss.

I'm not 100% sure, but I believe these regexes are only applied to single lines at a time, so `\s` will not accidentally match newline characters.

If so, we can change to:
```
^//\s*AssignAll disable\s*$
^//\s*AssignAll enable\s*$
```

If `SingleLineCommentTrivia` somehow allow newlines, then we can tighten it a bit more:

```
^//[^\S\r\n]*AssignAll disable[^\S\r\n]*$
^//[^\S\r\n]*AssignAll enable[^\S\r\n]*$
```
https://stackoverflow.com/a/3469155/134761

File: AssignAll/AssignAll.Test/UnitTests.cs [hunk 2]
```diff
@@ -118,28 +118,30 @@ public async Task EmptyCode_AddsNoDiagnostics()
             await VerifyCS.VerifyAnalyzerAsync(test);
         }
 
-        [Fact]
-        public async Task EnableAndDisableComments_EnablesAndDisablesAnalyzerForTextSpans()
+        [Theory]
+        [InlineData("// AssignAll")]
+        [InlineData("//AssignAll")]
```
Comment (angularsen) [inline AssignAll/AssignAll.Test/UnitTests.cs]: Add test cases on trailing whitespace + combinations with tab/regular whitespace and single/multiple spaces.

**Review by angularsen — CHANGES_REQUESTED** [review]
Very nice 👏 
Just a few suggestions.

File: AssignAll/AssignAll/AssignAllMembers/RegionsToAnalyzeProvider.cs (hunk 1 above)
Comment (anserg256) [reply AssignAll/AssignAll/AssignAllMembers/RegionsToAnalyzeProvider.cs]: Sure, extended the regex. I think we can assume that if there are line breaks in-between, it will not be considered as a single-line comment. I've tried to throw in some `\n` and `\r` in the test and it was just a compile error since it breaks the comment.

Comment (angularsen) [reply AssignAll/AssignAll/AssignAllMembers/RegionsToAnalyzeProvider.cs]: Great

**Comment by angularsen** [comment]
Very nice, nuget should be out shortly.

[Release 1.8.0 · angularsen/roslyn-analyzers](https://github.com/angularsen/roslyn-analyzers/releases/tag/1.8.0)
